tls/extensions/s2n_client_early_data_indication.c

```diff
@@ -72,6 +72,7 @@ static S2N_RESULT s2n_early_data_config_is_possible(struct s2n_connection *conn)
 static bool s2n_client_early_data_indication_should_send(struct s2n_connection *conn)
 {
     return s2n_result_is_ok(s2n_early_data_config_is_possible(conn))
+            && conn && conn->early_data_expected
```

Comment:
Shouldn't we update the `s2n_server_early_data_indication_should_send` function as well to check for the bitflag `conn->early_data_expected` being set ?

Reply:
The server considers conn->early_data_expected when setting S2N_EARLY_DATA_ACCEPTED, and the extension is sent based off of S2N_EARLY_DATA_ACCEPTED.

Reply:
I see but when in the handshake should `s2n_connection_set_early_data_expected` function be called (prior to early data extension) and how do we enforce that, as this is a pre-requisite required prior to the `s2n_early_data_accept_or_reject` function call.

Reply:
We need a single decision point where we either accept or reject the early data. If `early_data_expected` is not set at that decision point, we reject the early data. For the server, that decision point is `s2n_early_data_accept_or_reject`. If we accept the early data, we MUST send the extension. We can't accept the early data but not send the extension.